pandas/core/strings/accessor.py

```diff
@@ -1333,6 +1334,29 @@ def replace(self, pat, repl, n=-1, case=None, flags=0, regex=None):
                     )
                 warnings.warn(msg, FutureWarning, stacklevel=3)
             regex = True
+
+        # Check whether repl is valid (GH 13438, GH 15055)
```

Comment:
is this the only place this is done? can you make a function which does this

Reply:
> is this the only place this is done? 

it is. and hence the precurosr to avoid repeating the validation in the ArrowStringArray._str_replace method (which is not yet added)

> can you make a function which does this

this is that function. this function validates the input before dispatching to the relevant string kernel.

As we add more string methods to ArrowStringArray, we can move the validation from the string kernel to the accessor. If we find we have duplicated validation we could create a common validator function.